vm/src/obj/objint.rs

```diff
@@ -750,38 +753,30 @@ struct IntToByteArgs {
 }
 
 // Casting function:
-pub fn to_int(vm: &VirtualMachine, obj: &PyObjectRef, base: &BigInt) -> PyResult<BigInt> {
-    let base = match base.to_u32() {
-        Some(base) if base == 0 || (2..=36).contains(&base) => base,
-        _ => return Err(vm.new_value_error("int() base must be >= 2 and <= 36, or 0".to_owned())),
-    };
-
-    let bytes_to_int = |bytes: &[u8]| {
-        std::str::from_utf8(bytes)
-            .ok()
-            .and_then(|s| str_to_int(s, base))
```

Comment:
This looks like it's basically the same function as to_int_radix, is there a reason you can just call that with base as 10?

Reply:
base 10 is the most common case, so it will be worth to optimize it.

2 technical reasons:
- By passing constant, bytes_to_int (hopefully) will by optimized for 10; otherwise we can manually work for it later.
- `to_int_radix` process only string, bytes and bytearray. it skips other types.

Reply:
Hmm, ok. You could probably have to_int_radix return an actual python error in the `_` case, rather than checking in `int.__new__` and having an unreachable in to_int_radix.

Reply:
Oh, yep. Actually that was done seconds ago after checking priority of `val` and `base`

Reply:
Oh, I see that lol